src/types/misc.ts

```diff
@@ -6,3 +6,11 @@ export type StateSetter<T> = Dispatch<SetStateAction<T>>;
 export type IIndex<IndexType extends number | string, DataType> = {
   [Id in IndexType]: DataType;
 }
+
+// TODO A generic type for an interface containing a thing and ID for use in
+// state. This type maybe shouldn't contain the word "State"? ¯\_(ツ)_/¯
+// "Thing" instead?
```

Comment:
- [ ] Clarify comment: The purpose of this would be to merge many pieces of state which now represent the ID and the value as separate atoms.